csharp/src/Apache.Arrow/Ipc/ArrowReaderImplementation.cs

```diff
@@ -144,7 +154,7 @@ private void ReadDictionaryBatch(Flatbuf.DictionaryBatch dictionaryBatch, ByteBu
 
             if (dictionaryBatch.IsDelta)
             {
-                throw new NotImplementedException("Dictionary delta is not supported yet");
+                DictionaryMemo.AddDeltaDictionary(id, arrays[0], _allocator);
```

Comment:
Creating `arrays` for a delta dictionary causes needless memory allocation for  `ArrowArray`.
It's enough to create just `ArrayData` for the delta dictionary.
I plan to fix them in future PRs.

Reply:
I tested reading delta dictionaries by creating a test file `delta_dictionary_batch_test.batch` with python and reading `delta_dictionary_batch_test.batch` with C#, and it worked fine.

```
import pyarrow as pa

ty = pa.dictionary(pa.int8(), pa.utf8())
data = [["foo", "foo", None],
            ["foo", "bar", "foo"],
            ["foo", "bar"],
            ["foo", None, "bar", "quux"], 
            ["bar", "quux"],
            ]
batches = [
        pa.RecordBatch.from_arrays([pa.array(v, type=ty)], names=['dicts'])
        for v in data]
schema = batches[0].schema

def write_batches():
    options = pa.ipc.IpcWriteOptions(emit_dictionary_deltas=True)
    with pa.RecordBatchStreamWriter("./delta_dictionary_batch_test.batch", schema = schema, options = options) as writer:
        for batch in batches:
            writer.write_batch(batch)

st = write_batches()

import pyarrow as pa
```

Reply:
Note, once my integration change #10973 is merged, you can enable the dictionary integration tests to make sure dictionaries are working across C# and the other languages.